File: articles/security-failures-as-algorithmic-failures/prolog/security_failures.pl
weight(0.08). weight(0.10). weight(0.10). weight(0.10). weight(0.08). weight(0.08). weight(0.08).
weight(0.08). weight(0.08). weight(0.08). weight(0.06). weight(0.05). weight(0.03).
score(S) :- findall(W, weight(W), Ws), sum_list(Ws, Total), S is 0.65 * Total * 100.
:- initialization((score(S), write(S), nl, halt)).

File: articles/security-failures-as-algorithmic-failures/typescript/security_failures.ts
const weights: number[] = [0.08,0.10,0.10,0.10,0.08,0.08,0.08,0.08,0.08,0.08,0.06,0.05,0.03];
const score = weights.reduce((total, weight) => total + 0.65 * weight, 0) * 100;
console.log(score.toFixed(3));
